pkg/types/powervs/validation/machinepool.go

```diff
@@ -72,10 +72,20 @@ func ValidateMachinePool(p *powervs.MachinePool, fldPath *field.Path) field.Erro
 
 	// Validate SysType
 	if p.SysType != "" {
-		const sysTypeRegex = `^(?:e980|s922(-.*|))$`
-		// Allowing for a staging-only pattern of s922-* but not exposing here
-		if !regexp.MustCompile(sysTypeRegex).MatchString(p.SysType) {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("sysType"), p.SysType, "system type must be one of {e980,s922}"))
+		found := false
+		for i := range powervs.Regions {
+			for j := range powervs.Regions[i].SysTypes {
+				if powervs.Regions[i].SysTypes[j] == p.SysType {
+					found = true
+					break
+				}
+			}
+			if found {
+				break
+			}
+		}
```

Comment:
Why not create a `AllKnownSysTypes()` in `powervs_region.go` which returns a set so you can check for inclusion here?
```golang
func AllKownSysTypes() sets.Set[string] {
    sysTypes := sets.New[string]()
    for _, region := range Regions {
        sysTypes.Insert(region.SysTypes...)
    }
    return sysTypes
}
```
Then you can just:
```golang
if !AllKnownSysTypes().Has(p.SysType) {
    allErrs = append(allErrs, field.Invalid(fldPath.Child("sysType"), p.SysType, "unknown system type specified"))
}
```
If you want to be a bit more efficient you can also make it a variable instead of a function:
```golang
var AllKnownSysTypes = func() sets.Set[string] {
    sysTypes := sets.New[string]()
    for _, region := range Regions {
        sysTypes.Insert(region.SysTypes...)
    }
    return sysTypes
}
```

Reply:
Oh, I like it. Will update the code...

Reply:
Addressed in latest force-push. PTAL.